logstash-core/src/test/java/org/logstash/AccessorsTest.java

```diff
@@ -32,7 +32,7 @@
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertTrue;
 
-public class AccessorsTest {
+public class AccessorsTest extends RubyTestBase {
```

Comment:
these Java tests caused JRuby runtime initialization ...
-> we would lost control of booting a proper `RubyUtil.RUBY` global runtime

Reply:
> Would it be possible to have this behaviour powered by an annotation instead of subclassing?

yes it would be possible - there's simply an existing convention for some of the IR "ruby-aware" tests 
to use [`RubyEnvTestCase.java`](https://github.com/elastic/logstash/blob/v7.14.0/logstash-core/src/test/java/org/logstash/config/ir/RubyEnvTestCase.java#L33-L39) as a super-class, so here we're following the same ...

also considered the other option of having a JUnit style `@BeforeClass` boot but that still would depend on order or hooks since the `RubyEnvTestCase.java` is using one where it assumes `RubyUtil.RUBY` already booted.

> But I'm not sure that a superclass is the best path. What happens when a new test that uses jruby bits fails to subclass this testing base? Do we end up with wildly-broken behaviour that is easy to trace down, or subtly-broken behaviour that continues to work unless the right edge-case is hit? Is there anything we can do to ensure that we get the former instead of the latter?

Adding/changing an existing Java test (which would be using `org.jruby` bits directly or indirectly and forgot to sub-class or annotate) would lead to an [`AssertionError`](https://github.com/elastic/logstash/pull/13159/files#diff-0e1d049630fdb80df50e27e2bb79fd40c2c0b1ee28c614598e49a72931cfe5b4R59) when initializing the base class (or annotation) -> all classes that depend on the class than lead to a `NoClassDefFoundError` ... here's a [sample CI output where I forgot to sub-class one case](https://logstash-ci.elastic.co/job/elastic+logstash+pull-request+multijob-java-unit-tests/3689/consoleFull)

whether it's an annotation or sub-class does not make much of a difference (in terms of the failure, I think) ... 
if you feel it still provides readability to have an annotation instead I am happy to refactor :wink: